examples/matmul/run.cpp

```diff
@@ -288,7 +303,11 @@ fn main(
           localM[idx] = tileA[(threadRow + idx) * {{BK}} + dotIdx];
         }
         for (var idx: u32 = 0; idx < {{TN}}; idx++) {
-          localN[idx] = tileB[(threadCol + idx) * {{BK}} + dotIdx];
+          if ({{TRANSPOSE}}) {
+            localN[idx] = tileB[dotIdx * {{BN}} + threadCol + idx];
+          } else {
+            localN[idx] = tileB[(threadCol + idx) * {{BK}} + dotIdx];
+          }
```

Comment:
> Nice! That's a bit surprising - most examples I've seen seem to prefer the transposed forms of weights for efficiency, with both weights and input element-wise contributions into the dot product moving from left to right columns. This is already counter to the mathematical representation, so I'm surprised transposing the weights from that helps but there's probably something I'm missing.

The current transposed weights are suitable for cpu to prefetch data  as 1D array from DRAM.  
Tiled matrix multiplication needs to load data as 2D array, then the advantage is diminished.
So there will probably not be much difference in loading between transposed case and not-transposed case.

Next, this PR changes the access of shared memory like this.
This may have reduced bank collisions and therefore improved performance.

https://www.vrushankdes.ai/diffusion-policy-inference-optimization/part-ii---cuda-kernel-optimization-tips#compute-optimizations-

If memory padding improves performance, bank collisions may be the cause.

> I would suggest just having a different shader including comments on the diffs instead of relying on the removeUnnecessaryIfStatements. This makes it more explicit. The regex transformation works of course, but if we start composing these regex manipulations too deeply I think things will start to getting painful to debug + understand what's going on.

I included the removeUnnecessaryIfStatements to check the difference in code, but I will create a new kernel function not to use it.